### PR TITLE
preserve order of items in state when setting a node selected

### PR DESCRIPTION
**Pull request by Blackbaud-LuisBello**

### CODE REVIEW

File: src/modules/tree-view/tree-view.component.ts [hunk 1]
```diff
@@ -122,13 +122,13 @@ export class SkyTreeViewComponent implements OnInit, AfterContentInit {
 
   private isLeaf(nodeId: string, nodes: Array<TreeNodeModel>) {
     let isLeaf = true;
-    for (let i = 0; i < nodes.length; i++) {
-      let node = nodes[i];
-      if (node.parent != null && nodeId === node.parent.id) {
+    nodes.forEach(node => {
+      if (node.parent && node.parent.id === nodeId) {
         isLeaf = false;
-        break;
```
Comment (blackbaud-brandonstirnaman) [inline src/modules/tree-view/tree-view.component.ts]: Don't make changes like this for aesthetic reasons. This changes the behavior. A `break` inside the `for` loop will actually stop the looping. There is no way to stop a javascript `.forEach` from continuing to iterate.

**Comment by Blackbaud-LuisBello** [comment]
updated

File: src/modules/tree-view/tree-view-node.component.ts [hunk 2]
```diff
@@ -46,7 +46,11 @@ export class SkyTreeViewNodeComponent {
   }
 
   public get enabled() {
-    return this.disableParents ? !this.hasSelectedChildren(this.node) : true;
+    if (!this.node.enabled) {
+      return false;
+    } else {
```
Comment (blackbaud-brandonstirnaman) [inline src/modules/tree-view/tree-view-node.component.ts]: this doesn't need an else block.

File: src/modules/tree-view/tree-node.model.ts [hunk 3]
```diff
@@ -5,13 +5,17 @@ export class TreeNodeModel {
...
   public isExpanded = false;
+  public enabled = true;
   public data: any;
 
   constructor(data: any = null) {
     if (data != null) {
       this.id = data.id;
       this.name = data.name;
       this.parent = data.parent;
+      this.enabled = data.enabled != null ? data.enabled : true;
```
Comment (blackbaud-brandonstirnaman) [inline src/modules/tree-view/tree-node.model.ts]: `!= null` isn't necessary here

**Pull request merged**